util/sponsor/auth.go

```diff
@@ -52,6 +60,8 @@ func ConfigureSponsorship(token string) error {
 		}
 	}
 
+	Token = token
```

Comment:
Not sure. This may be a broken token and is potentially used in other parts of the application.

Reply:
> This may be a broken token

Yes, and I want to reflect this in the UI. See screenshot "broken token dialog"

> is potentially used in other parts of the application.

No it isn't.

Reply:
telemetry.go

Reply:
Access is always checked by `IsAuthorized[ForApi]` first. Existence of token is currently not treated as valid sponsorship. We could make Token private and only return it via getter that checks internally.